aws_lambda_powertools/shared/functions.py

```diff
@@ -275,13 +275,11 @@ def abs_lambda_path(relative_path: str = "") -> str:
         If the path is empty, it will return the current working directory.
     """
     # Retrieve the LAMBDA_TASK_ROOT environment variable or default to an empty string
-    current_working_directory = os.environ.get("LAMBDA_TASK_ROOT", "")
+    current_working_directory = os.environ.get("LAMBDA_TASK_ROOT", "") or str(Path.cwd())
 
-    # If LAMBDA_TASK_ROOT is not set, use the current working directory
-    if not current_working_directory:
-        current_working_directory = str(Path.cwd())
+    return str(Path(current_working_directory, relative_path))
 
-    # Combine the current working directory and the relative path to get the absolute path
-    absolute_path = str(Path(current_working_directory, relative_path))
 
-    return absolute_path
+def sanitize_xray_segment_name(name: str) -> str:
+    _common_invalid_name_characters = "?;*()!$~^<>"
+    return "".join(char for char in name if char not in _common_invalid_name_characters)
```

Comment:
Actually, now that I'm thinking on the code, I believe this would be faster:

```python
INVALID_CHARS_PATTERN = r"[?;*()!$~^<>]"
def sanitize_xray_segment_name(name: str) -> str:
    return re.sub(invalid_chars_pattern, "", name)
 ```

Reply:
I'm not sure about this. With the `re` approach, we need to import the module and the execution time of `re.sub` might be slightly slower compared to a simple iteration.

Look at these very simple example.

#### re_test.py
```python
import re

INVALID_CHARS_PATTERN = r"[?;*()!$~^<>]"
def sanitize_xray_segment_name(name: str) -> str:
    return re.sub(INVALID_CHARS_PATTERN, "", name)

name = sanitize_xray_segment_name("ijheirkhwekwd098;92093kjwd()!kjnsdkjwhjewjieowueuwuieuwioe~^<>")
```
![image](https://github.com/aws-powertools/powertools-lambda-python/assets/4295173/c9b21ba3-ca93-4789-a62f-ad32831e6c3d)


#### for_test.py
```python
INVALID_CHARS_PATTERN = r"[?;*()!$~^<>]"
def sanitize_xray_segment_name(name: str) -> str:
    return "".join(char for char in name if char not in INVALID_CHARS_PATTERN)

name = sanitize_xray_segment_name("ijheirkhwekwd098;92093kjwd()!kjnsdkjwhjewjieowueuwuieuwioe~^<>")
```
![image](https://github.com/aws-powertools/powertools-lambda-python/assets/4295173/d0e93f16-61b2-4567-a956-f513f60b4208)

I think we can use iteration instead of regex.

Reply:
Sure, but can you run the example 1000 times? It's unrealistic to only measure the cold start :)

Reply:
Yeah, sure! Using hyperfine..

```sh
❯ hyperfine --runs 1000  'python for_test.py' -i
Benchmark 1: python for_test.py
  Time (mean ± σ):       7.4 ms ±   0.7 ms    [User: 5.8 ms, System: 1.5 ms]
  Range (min … max):     6.3 ms …  15.6 ms    1000 runs
```
 
```sh
❯ hyperfine --runs 1000  'python re_test.py' -i
Benchmark 1: python re_test.py
  Time (mean ± σ):      11.5 ms ±   0.8 ms    [User: 9.4 ms, System: 2.0 ms]
  Range (min … max):    10.0 ms …  20.7 ms    1000 runs
```

Reply:
Code:

```python
import re

INVALID_CHARS_PATTERN = "?;*()!$~^<>"
INVALID_CHARS_RE_PATTERN = r"[?;*()!$~^<>]"


def using_for(name: str) -> str:
    return "".join(char for char in name if char not in INVALID_CHARS_PATTERN)


def using_re(name: str) -> str:
    return re.sub(INVALID_CHARS_RE_PATTERN, "", name)


if __name__ == "__main__":
    import timeit

    print(
        timeit.timeit("using_for('ijheirkhwekwd098;92093kjwd()!kjnsdkjwhjewjieowueuwuieuwioe~^<>')", globals=locals()),
    )
    print(timeit.timeit("using_re('ijheirkhwekwd098;92093kjwd()!kjnsdkjwhjewjieowueuwuieuwioe~^<>')", globals=locals()))

```

Results:

```
✦ ❯ python bench.py
2.6501595829613507
1.1926250829128549
```

using re.sub vs iteration is **at least** two times faster

Reply:
Thank you @rubenfonseca! I changed the function.